src/LcovGeneratorTests/LcovGeneratorTests.cs

```diff
@@ -0,0 +1,83 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Spekt.Vstest.Coverage;
+using Spekt.Vstest.Coverage.LcovGenerator;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace LcovGeneratorTests
+{
+    [TestClass]
+    public class LcovGeneratorTests
+    {
+
+        public LcovGeneratorTests()
+        {
+            LcovModel lcovModel = new LcovModel();
+        }
+
+        [TestMethod]
+        public void GeneratorShouldReturnCorrectDictionary()
+        {
+            CodeCoverageDataModel codeCoverageDataModel = new CodeCoverageDataModel();
+            Generator generator = new Generator();
+            codeCoverageDataModel.Sources.Add("s1.cs");
+            codeCoverageDataModel.Sources.Add("s2.cs");
+            Range range1 = new Range(5, 6, true, 0);
+            Range range2 = new Range(9, 12, true, 1);
+
+            codeCoverageDataModel.Ranges.Add(range1);
+            codeCoverageDataModel.Ranges.Add(range1);
+            codeCoverageDataModel.Ranges.Add(range2);
+
+            Dictionary<string, Dictionary<int, int>> expecedLcovModelData = new Dictionary<string, Dictionary<int, int>>();
+            expecedLcovModelData.Add("s1.cs", new Dictionary<int, int> { {5,2 }, { 6, 2} });
+            expecedLcovModelData.Add("s2.cs", new Dictionary<int, int> { { 9, 1 }, { 10, 1 }, { 11, 1}, { 12, 1} });
+
+            var actualLcovModelData = generator.ModelConverter(codeCoverageDataModel);
+           
+            foreach(var modelData in expecedLcovModelData)
+            {
+                CollectionAssert.AreEqual(expecedLcovModelData[modelData.Key], actualLcovModelData.Data[modelData.Key]);
+            }      
+        }
+
+        [TestMethod]
+        public void LcovModelToStringShouldWriteToFile()
+        {
+            CodeCoverageDataModel codeCoverageDataModel = new CodeCoverageDataModel();
+            Generator generator = new Generator();
+            codeCoverageDataModel.Sources.Add("s1.cs");
+            codeCoverageDataModel.Sources.Add("s2.cs");
+            Range range1 = new Range(5, 6, true, 0);
+            Range range2 = new Range(9, 12, true, 1);
+
+            codeCoverageDataModel.Ranges.Add(range1);
+            codeCoverageDataModel.Ranges.Add(range1);
+            codeCoverageDataModel.Ranges.Add(range2);
+
+            Dictionary<string, Dictionary<int, int>> expecedLcovModelData = new Dictionary<string, Dictionary<int, int>>();
+            expecedLcovModelData.Add("s1.cs", new Dictionary<int, int> { { 5, 2 }, { 6, 2 } });
+            expecedLcovModelData.Add("s2.cs", new Dictionary<int, int> { { 9, 1 }, { 10, 1 }, { 11, 1 }, { 12, 1 } });
+
+            var actualLcovModelData = generator.ModelConverter(codeCoverageDataModel);
+
+            var actualText = actualLcovModelData.ToString();
+            System.Console.WriteLine(actualText);
+
+            string expectedText = "SF:s1.cs" + Environment.NewLine +
```

Comment:
Move the expected text to file. To make it readable. Tests should help us to doc.